Assets/Scripts/Combat/Abilities/SliderHandle.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SliderHandle : MonoBehaviour
+{
+    [SerializeField] private GameObject slider;
+    [SerializeField] private GameObject HitArea;
+    [SerializeField] private GameObject start;
+    [SerializeField] private GameObject end;
+
+    private RectTransform HitTransform;
+
+    private BoxCollider2D SliderCollider;
+    private BoxCollider2D GoodCollider;
+    private BoxCollider2D PerfectCollider;
+
+    private float SliderSpeed = 0.7f;
+    private int NumberOfClicks;
+    private int MaxClicks = 3;
+    private int GoodClicks;
+    private int PerfectClicks;
+    private bool arrived = false;
+    private bool isDone = false;
+
+    private enum Click { None, Miss, Good, Perfect } 
+
+    private Click click = Click.None;
+
+    private void Start()
+    {
+        //Get Components
+        HitTransform = HitArea.GetComponent<RectTransform>();
+        SliderCollider = slider.GetComponent<BoxCollider2D>();
+        GoodCollider = HitArea.GetComponent<BoxCollider2D>();
+        PerfectCollider = HitArea.transform.GetChild(0).GetComponent<BoxCollider2D>();
+
+        enabled = false;
```

Comment:
What/Where is this boolean used for

Reply:
It disables / enables the script. 
I just put it in a function so it's clearer. 
I need it cause otherwise the Click variables increment every time you press even when I don't need them to.